presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewCandidateExtractor.java

```diff
@@ -22,41 +22,49 @@
 import com.facebook.presto.sql.parser.SqlParser;
 import com.facebook.presto.sql.tree.Query;
 import com.google.common.collect.ImmutableList;
-import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableMap;
 import org.testng.annotations.Test;
 
 import java.util.List;
-import java.util.Set;
+import java.util.Map;
 
 import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
 import static com.facebook.presto.testing.assertions.Assert.assertEquals;
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.lang.String.format;
 
 public class TestMaterializedViewCandidateExtractor
```

Comment:
Should be add few tests with With clause?

Reply:
I'm totally removing the `candidateExtractor` in https://github.com/prestodb/presto/pull/17912 which I aim to be the next PR I merge after this one. For now I added 1 test, but let me know if you still think I should be more thorough in light of the upcoming removal.